sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -203,6 +203,7 @@ def _transform(self, X, fitting):
             feature_names = []
             vocab = {}
         else:
+            check_is_fitted(self, ["feature_names_", "vocabulary_"])
```

Comment:
I have inserted the check into `_transform()` instead of `transform()`, because `transform(X)` is only a proxy for `_transform(X, fitting=False)`.

Reply:
in terms of readability, I think it makes more sense to have this in `transform` rather than here.